js/src/components/tree-select-control/tags.js

```diff
@@ -2,119 +2,64 @@
  * External dependencies
  */
 import { __, sprintf } from '@wordpress/i18n';
-import { Button } from '@wordpress/components';
-import { Icon, cancelCircleFilled } from '@wordpress/icons';
-import { createElement, Component, Fragment } from '@wordpress/element';
-import { findIndex } from 'lodash';
-import PropTypes from 'prop-types';
 import { Tag } from '@woocommerce/components';
 
-/**
- * Internal dependencies
- */
-// import Tag from '../tag';
-
 /**
  * A list of tags to display selected items.
+ *
+ * @param {Object} props The component props
+ * @param {Object[]} props.tags The tags
+ * @param {Function} props.onChange The method called when a tag is removed
+ * @param {boolean} props.disabled True if the plugin is disabled
  */
-class Tags extends Component {
-	constructor( props ) {
-		super( props );
-		this.removeAll = this.removeAll.bind( this );
-		this.removeResult = this.removeResult.bind( this );
-	}
-
-	removeAll() {
-		const { onChange } = this.props;
-		onChange( [] );
+const Tags = ( { tags, disabled, onChange = () => {} } ) => {
+	if ( ! tags.length ) {
+		return null;
 	}
 
-	removeResult( key ) {
+	/**
+	 * Callback to remove a Tag.
+	 * The function is defined this way because in the WooCommerce Tag Component the remove logic
+	 * is defined as `onClick={ remove(key) }` hence we need to do this to avoid calling remove function
+	 * on each render.
+	 *
+	 * @param {string} key The key for the Tag to be deleted
+	 */
+	const remove = ( key ) => {
 		return () => {
-			const { selected, onChange } = this.props;
-			const i = findIndex( selected, { key } );
-			onChange( [
-				...selected.slice( 0, i ),
-				...selected.slice( i + 1 ),
-			] );
+			if ( disabled ) {
+				return;
+			}
+			const i = tags.findIndex( ( tag ) => tag.id === key );
+			onChange( [ ...tags.slice( 0, i ), ...tags.slice( i + 1 ) ] );
```

Comment:
Fixed here: https://github.com/woocommerce/google-listings-and-ads/pull/1243/commits/1113347cbfdf399ec6fac82f48323b8b32d9108f

It's the same as just suggestion but using `tag` instead of `el`